TACC/Frontera-Portal
# task/FP-791 shared workspace file listing, manage team members

## Overview: ##

Exposes metadata in Shared Workspaces view

## PR Status: ##


## Related Jira tickets: ##

* [FP-213](https://jira.tacc.utexas.edu/browse/FP-213)
* [FP-791](https://jira.tacc.utexas.edu/browse/FP-791)
* [FP-792](https://jira.tacc.utexas.edu/browse/FP-792)

## Summary of Changes: ##

- Metadata sagas and reducers
- Refactor dom for datafiles components (search bar goes into DataFilesListing component)
- Metadata in shared workspaces
- Ability to manage members (add, remove, transfer ownership)
- Backend updates to translate Shared Workspace concepts to Projects ("owner" and "edit" access are "pi" and "co_pi")
- Backend updates to support transferring ownership

## Testing Steps: ##
1. Get a second user using one of the test accounts on your local system
2. Create a project
3. Make sure you can add the test user to your project
4. Make sure you can transfer ownership back and forth
5. Make sure you can remove users

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/101682330-14162d80-3a29-11eb-9f02-7acc86d6d1e0.png)

![image](https://user-images.githubusercontent.com/17181582/101937327-56fe0f80-3ba7-11eb-9cb9-7f6297ca8afc.png)


## Notes: ##

Please review task/FP-789 first, and then I will update from that branch.
